sources/packages/web/src/views/aest/student/StudentRestrictions.vue

```diff
@@ -1,229 +1,28 @@
   props: {
     studentId: {
       type: Number,
       required: true,
     },
   },
-  setup(props) {
-    const studentRestrictions = ref();
-    const { dateOnlyLongString } = useFormatters();
-    const showModal = ref(false);
-    const viewRestriction = ref(
-      {} as ModalDialog<RestrictionDetailAPIOutDTO | boolean>,
-    );
-    const addRestriction = ref(
-      {} as ModalDialog<AssignRestrictionAPIInDTO | boolean>,
-    );
-    const studentRestriction = ref();
-    const snackBar = useSnackBar();
-
-    const loadStudentRestrictions = async () => {
-      studentRestrictions.value =
-        await RestrictionService.shared.getStudentRestrictions(props.studentId);
-    };
-
-    const viewStudentRestriction = async (restrictionId: number) => {
-      studentRestriction.value =
-        await RestrictionService.shared.getStudentRestrictionDetail(
-          props.studentId,
-          restrictionId,
-        );
-      studentRestriction.value.createdAt = dateOnlyLongString(
-        studentRestriction.value.createdAt,
-      );
-      if (studentRestriction.value.updatedAt) {
-        studentRestriction.value.updatedAt = dateOnlyLongString(
-          studentRestriction.value.updatedAt,
-        );
-      }
-      const viewStudentRestrictionData =
-        await viewRestriction.value.showModal();
-      if (viewStudentRestrictionData) {
-        resolveRestriction(
-          viewStudentRestrictionData as RestrictionDetailAPIOutDTO,
-        );
-      }
-    };
-
-    const resolveRestriction = async (data: RestrictionDetailAPIOutDTO) => {
-      try {
-        const payload = {
-          noteDescription: data.resolutionNote,
-        } as ResolveRestrictionAPIInDTO;
-        await RestrictionService.shared.resolveStudentRestriction(
-          props.studentId,
-          data.restrictionId,
-          payload,
-        );
-        await loadStudentRestrictions();
-        snackBar.success(
-          "The given restriction has been resolved and resolution notes added.",
-        );
-      } catch {
-        snackBar.error("Unexpected error while resolving the restriction.");
-      }
-    };
-
-    const addStudentRestriction = async () => {
-      const addStudentRestrictionData = await addRestriction.value.showModal();
-      if (addStudentRestrictionData) {
-        createNewRestriction(
-          addStudentRestrictionData as AssignRestrictionAPIInDTO,
-        );
-      }
-    };
-
-    const createNewRestriction = async (data: AssignRestrictionAPIInDTO) => {
-      try {
-        await RestrictionService.shared.addStudentRestriction(
-          props.studentId,
-          data,
-        );
-        await loadStudentRestrictions();
-        snackBar.success("The restriction has been added to student.");
-      } catch {
-        snackBar.error("Unexpected error while adding the restriction.");
-      }
-    };
-
-    onMounted(async () => {
-      await loadStudentRestrictions();
-    });
+  setup() {
```

Comment:
Is there a need to have the `setup` at all?

Reply:
I am not sure of another way to return the `StudentRestrictions` component so that it can be used in the template.

Reply:
you don' need to return the component

Reply:
Got it. Thanks